### PR TITLE
Refactor: sign out button

### LINKED ISSUE #1519: Refactor sign out (Login.gov) copy/button in Enrollment Success
Follow up to #1508 and part of finishing that refactor per #1442, we need to address the final usage of `viewmodels.Button.logout()`.

## Acceptance Criteria

- [ ] We can entirely remove the `from benefits.core import viewmodels` from `enrollment/views.py`
- [ ] The sign out (Login.gov) button shows correctly for flows that support it (`senior`)
- [ ] The sign out (Login.gov) button does not show for flows that don't support it (`mst_courtesy_card`, `veteran`)

## Additional context

image

### PR DESCRIPTION
**Pull request by thekaveman**
Closes #1519

Needed for #1520 / #1531

Part of #1442

## Sign out link

image

## Sign out button

image

### CODE REVIEW

File: tests/pytest/conftest.py [hunk 1]
```diff
@@ -44,12 +44,16 @@ def model_PemData():
...
-        sign_out_button_label="Sign out",
+        sign_out_button_template="core/includes/sign-out-button--senior.html",
+        sign_out_link_template="core/includes/sign-out-link--senior.html",
         client_name="Client",
         client_id="1234",
         authority="https://example.com",
     )
 
+    assert not auth_provider.supports_claims_verification
+    assert auth_provider.supports_sign_out
```
Comment (angela-tran) [inline tests/pytest/conftest.py:55]: What is the reason for having these assertions here rather than in the test?

**Review by angela-tran — COMMENTED** [review]
The refactor looks great! Just a minor question about test assertions

File: tests/pytest/conftest.py [hunk 2]
```diff
@@ -59,16 +63,22 @@ def model_AuthProvider_with_verification(model_AuthProvider):
     model_AuthProvider.claim = "claim"
     model_AuthProvider.save()
 
+    assert model_AuthProvider.supports_claims_verification
```
Comment (angela-tran) [inline tests/pytest/conftest.py:66]: Same question as above

File: tests/pytest/conftest.py [hunk 3]
```diff
@@ -59,16 +63,22 @@ def model_AuthProvider_with_verification(model_AuthProvider):
...
     model_AuthProvider.scope = "scope"
     model_AuthProvider.claim = "claim"
-    model_AuthProvider.sign_out_button_label = None
+    model_AuthProvider.sign_out_button_template = None
+    model_AuthProvider.sign_out_link_template = None
     model_AuthProvider.save()
 
+    assert model_AuthProvider.supports_claims_verification
+    assert not model_AuthProvider.supports_sign_out
+
```
Comment (angela-tran) [inline tests/pytest/conftest.py:81]: Same question as above

File: tests/pytest/conftest.py [hunk 4]
```diff
@@ -78,16 +88,22 @@ def model_AuthProvider_without_verification(model_AuthProvider):
     model_AuthProvider.claim = None
     model_AuthProvider.save()
 
+    assert not model_AuthProvider.supports_claims_verification
```
Comment (angela-tran) [inline tests/pytest/conftest.py:91]: Same question as above

File: tests/pytest/conftest.py [hunk 5]
```diff
@@ -78,16 +88,22 @@ def model_AuthProvider_without_verification(model_AuthProvider):
...
 def model_AuthProvider_without_verification_no_sign_out(model_AuthProvider):
     model_AuthProvider.scope = None
     model_AuthProvider.claim = None
-    model_AuthProvider.sign_out_button_label = None
+    model_AuthProvider.sign_out_button_template = None
+    model_AuthProvider.sign_out_link_template = None
     model_AuthProvider.save()
 
+    assert not model_AuthProvider.supports_claims_verification
+    assert not model_AuthProvider.supports_sign_out
```
Comment (angela-tran) [inline tests/pytest/conftest.py:105]: Same question as above

File: tests/pytest/conftest.py (hunk 1 above)
Comment (thekaveman) [reply tests/pytest/conftest.py:55]: Yeah there wasn't any explicit tests for these models, I could add them instead of asserting here.

Comment (angela-tran) [reply tests/pytest/conftest.py:55]: Oh I see, `supports_sign_out` is not on `EligibilityVerifier` anymore and that's the model we have tests for

Comment (angela-tran) [reply tests/pytest/conftest.py:55]: I'm ok with leaving them here

**Review by angela-tran — APPROVED** [review]

File: tests/pytest/conftest.py (hunk 1 above)
Comment (thekaveman) [reply tests/pytest/conftest.py]: I guess it just felt a little weird writing tests for the fixtures the tests are creating... 😆 

But we already do that, and the tests confirm the behavior in our standard way, so good call.

**Review by angela-tran — APPROVED** [review]
🚀

**Pull request merged**